### PR TITLE
More parallelism timeouts strict local backend heap size

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by danking** [comment]
This now also includes https://github.com/hail-is/hail/pull/13123

**Review by daniel-goldstein — APPROVED** [review]
Approving as all the related PRs that were rolled into this were already approved.

File: hail/python/hail/matrixtable.py [hunk 1]
```diff
@@ -1941,7 +1941,7 @@ def unfilter_entries(self):
         entry_ir = hl.if_else(
             hl.is_defined(self.entry),
             self.entry,
-            hl.literal(hl.Struct(**{k: hl.missing(v.dtype) for k, v in self.entry.items()})))._ir
+            hl.struct(**{k: hl.missing(v.dtype) for k, v in self.entry.items()}))._ir
```
Comment (danking) [inline hail/python/hail/matrixtable.py:1944]: this avoids an `eval` just to get `None` for the missing (which is the converted back to a missing IR)

File: hail/python/test/hail/experimental/test_annotation_db.py [hunk 2]
```diff
@@ -10,9 +10,8 @@ def db_json(init_hail):
         backend = hl.current_backend()
         if isinstance(backend, ServiceBackend):
             backend.batch_attributes = dict(name='setupAnnotationDBTests')
-        t = hl.utils.range_table(10)
-        t = t.key_by(locus=hl.locus('1', t.idx + 1))
-        t = t.annotate(annotation=hl.str(t.idx))
+        t = hl.utils.genomic_range_table(10)
+        t = t.annotate(annotation=hl.str(t.locus.position - 1))
```
Comment (danking) [inline hail/python/test/hail/experimental/test_annotation_db.py:14]: this avoids a useless shuffle

File: hail/python/test/hail/methods/test_impex.py [hunk 3]
```diff
@@ -38,19 +39,6 @@ class VCFTests(unittest.TestCase):
     def test_info_char(self):
         self.assertEqual(hl.import_vcf(resource('infochar.vcf')).count_rows(), 1)
 
-    def test_import_export_same(self):
```
Comment (danking) [inline hail/python/test/hail/methods/test_impex.py:41]: moved outside of the unittest class so that I can use pytest parameterization.

**Pull request merged**